### ISSUE TITLE
Not possible to crop and resize with Dolby Vision Source

### ISSUE THREAD

**Issue by Thomas-MB** [staxrip/staxrip#1304]
Hi,

I have an UHD source with Dolby Vision metadata and want to reencode it including cropping and resizing to 1080p.
So, the source has 3840x2160, cropping was automaticyll set to 276/276 (top/bottom) while the cropping values of the rpu written in the metadata json are 276/277.

If I'm activating the resize checkbox and move the slider to 1080 for the width it sets automatically the height to 808. The auto crop values as well as the calculated height  are as expected, but because of the minimal aspect ratio error of -0,50% there is a warning, which can't be ignored: "Resizing would interfere with the DV Metadata, Disable Resize or the DV RPU File."

So, neither I want to drop resizing nor dv. I could drop the cropping to get an 0% error, but that's also no way to go. From my perspective this behaviour of StaxRip (I'm using 2.37.5) is a bug, because I did very much encodings with 2.31.0 and doing the dv stuff afterwards with [DDVT](https://forum.doom9.org/showthread.php?t=183479), which produced good results. I did also very much tests and used my OLED which can handle dv and gives me a hint, when it plays a video with dv. And for sure, my encodings mostly have a slightly aspect ratio error.

For now, I would be happy to know how I can disable this warning (I couldn't find an option in the settings).
For future, I would like to handle the aspect ratio error same as before (within the configured boundary - per default there is an max aspect error of 2 defined).

Best regards
Thomas

**Comment by Dendraspis** [comment]
The warning cannot be disabled - you have to set a proper resolution.

You didn't mention the resolution, that you have tried to run, but accordning to the crop values, you should have end up with `3840x1608`, which sould result in `1920x804` when you move the slider to the left:

![image](https://github.com/staxrip/staxrip/assets/50659978/61ac0a97-c971-4833-9d31-407bcfd12a20)
![image](https://github.com/staxrip/staxrip/assets/50659978/105ab7c9-3217-4e62-9725-7edb1b1e142b)


If that does not work for you, for whatever reason, use the edit fields to put the width/height directly in with your keyboard.


> For future, I would like to handle the aspect ratio error same as before (within the configured boundary - per default there is an max aspect error of 2 defined).

Not possible, because that would ruin the outcome.

> I did very much encodings with 2.31.0 and doing the dv stuff afterwards with [DDVT](https://forum.doom9.org/showthread.php?t=183479), which produced good results

In this case the result could have gone wrong...

**Comment by Thomas-MB** [comment]
Hi @Dendraspis,

thanks for your reply.

You assumed correctly. Cropped image is 3840x1608 and sliding to 1920 (I wrote 1080, but meant 1080p...) results in height 808. When setting height manually to 804 it's not divisible by 8 anymore. I'm not sure which consequences this (not disivible by 8) has. I assume it has a little bit bad influence on the compression. Maybe it could result in problems for players? So, is it really a good idea to drop the mod 8?

How can this little aspect error (in my case -0,5%) "ruin the outcome"?

**Comment by Dendraspis** [comment]
Ah, I guess you still use old settings and have `Output Mod` set to `8`, right?  😉
Since a few versions the new default is `2`.

Back in the days `Mod 8` was used to make sure every app and player can handle the video, but meanwhile it is not necessary anymore unless you have old apps you wanna use or some weird specific requirements.

**Comment by Thomas-MB** [comment]
> Ah, I guess you still use old settings and have Output Mod set to 8, right? 😉
That's right. I moved my template to the new version. I will adjust this 👍 

> The metadata would not fit your video anymore.
As I said, I didn't notice any artifacts yet using DDVT afterwards mostly having a slightly aspect ratio error, but I will ensure to get zero aspect ratio errors for dv encodings from now on 👍

**Comment by Dendraspis** [comment]
> As I said, I didn't notice any artifacts yet using DDVT afterwards mostly having a slightly aspect ratio error

In this case it was only 1px over the whole height, so it was not perfect, but for most people ignorable. And there won't be artifacts, it's just the pixel changes (from the metadata), that are not accurate then, mostly noticeable in small bright parts like stars.

Have fun...

**Comment by Thomas-MB** [comment]
Sometimes ignorance is blessing. I know exactly what I do next time watching my old encodes. I'm looking into the stars...
